plugin/restore/mysql/mysql.go

```diff
@@ -50,9 +59,65 @@ func New(driver *mysql.Driver, instance *mysqlutil.Instance, connCfg db.Connecti
 	}
 }
 
-// replayBinlog restores the database using incremental backup in time range of [config.Start, config.End).
-func replayBinlog(ctx context.Context, config api.BinlogInfo, binlogDir string) error {
-	return fmt.Errorf("Unimplemented")
+// ReplayBinlog replays the binlog for `originDatabase` from `startBinlogInfo.Position` to `targetTs`.
+func (r *Restore) replayBinlog(ctx context.Context, originalDatabase, pitrDatabase string, startBinlogInfo api.BinlogInfo, targetTs int64) error {
+	if err := r.incrementalFetchAllBinlogFiles(ctx); err != nil {
+		return err
+	}
+
+	replayBinlogPaths, err := getBinlogReplayList(startBinlogInfo, r.binlogDir)
+	if err != nil {
+		return err
+	}
+	stopDateTime := getDateTime(targetTs)
+
```

Comment:
Let's add a comment saying something like

"Extract the SQL statements from the binlog and pipe/replay them to the pitrDatabase via the mysql client"